docs/arch/adr-002.md

```diff
@@ -0,0 +1,38 @@
+# ADR 2: Code Generation Process
+
+## Context
+
+We will be generating a number of classes as per [ADR 1](adr-001.md).
+
+Generated classes should be generated during the build process and not committed to GitHub.
+This ensures we have reproducable builds and makes it clear that generated classes are not to be edited manually.
+
+We wish to publish articats to the central maven repository.
+This does not necessarily limit us to working with Maven, but it encourages us to stick it.
+
+Using a compiled language to run the code generation would require two passes of compilation
+(or another/project module) - one to compile the codegen scripts, another to run it.
+
+## Decision
+
+We will use ruby to write the code generation scripts.
+
+We will run them using the [jruby-maven-plugins](https://github.com/torquebox/jruby-maven-plugins) as part of the
+`generate-sources` phase of the
+[maven lifecycle](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html)
```

Comment:
We should also reference 2 pass comp here as it fits the bill. This isn't just about the ruby phase of generation, is it?

Reply:
Or are we just using execution stages to handle it, if so we should define that?

Reply:
So, ideally this would have been written at the time I made this decision. One of the factors I used in deciding to use a scripting language was to avoid two passes. That we have added it later was not relevant to this decision